src/platform/plugins/shared/unified_doc_viewer/public/components/doc_viewer_traces_overview/sub_components/trace_summary.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { TRANSACTION_NAME_FIELD } from '@kbn/discover-utils';
+import { EuiHorizontalRule } from '@elastic/eui';
+import React, { useState, useEffect } from 'react';
+import { useTransactionContext } from '../hooks/use_transaction';
+import { FieldWithActions } from './field_with_actions/field_with_actions';
+import { FieldConfiguration } from '../resources/get_field_configuration';
+
+export interface TraceSummaryProps {
+  fieldId: string;
+  fieldConfiguration: FieldConfiguration;
+}
+
+export function TraceSummary({ fieldConfiguration, fieldId }: TraceSummaryProps) {
+  const { transaction, loading } = useTransactionContext();
+  const [fieldValue, setFieldValue] = useState(fieldConfiguration.value);
+  const isTransactionNameField = fieldId === TRANSACTION_NAME_FIELD;
+  const displayLoading = isTransactionNameField && !fieldValue;
+
+  useEffect(() => {
+    if (isTransactionNameField && !fieldValue && transaction && !loading) {
+      setFieldValue(transaction.name);
+    }
+  }, [transaction, fieldId, loading, fieldValue, isTransactionNameField]);
+
+  if ((!displayLoading && !fieldValue) || (displayLoading && !loading && !fieldValue)) {
```

Comment:
I feel very confused by looking at all this conditions.

Isn't it the same thing?
```suggestion
  if (!fieldValue && !displayLoading) {
    return null;
  }
```

Reply:
I included a brief explanation in [this comment](https://github.com/elastic/kibana/pull/210612#discussion_r1959516134) about the reasoning behind it and renamed the variable for better clarity.

Please let me know if this makes it clearer.